pulsar-client-schema/src/main/java/org/apache/pulsar/client/api/schema/GenericRecord.java

```diff
@@ -0,0 +1,51 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pulsar.client.api.schema;
+
+import java.util.List;
+
+/**
+ * An interface represents a message with schema.
+ */
+public interface GenericRecord {
+
+    /**
+     * Returns the list of fields associated with the record.
+     *
+     * @return the list of fields associated with the record.
+     */
+    List<Field> getFields();
```

Comment:
While it might not be required for JDBC integration, in this interface we should support the notion of nested objects. 
I'd suggest to try to mimic the interface of `JsonObject` : https://docs.oracle.com/javaee/7/api/javax/json/JsonObject.html

Reply:
Also, I think it would be useful to have the `getAsString()`, `getAsLong()` to avoid checking and casting for application

Reply:
I didn't do getAsString, getAsLong at this PR, because it is better to have primitive schemas (e.g. int, long, double, float, string, boolean). this generic interfaces allows unblocking JDBC connector first.

Reply:
Sure, we can do the minimum required by JDBC now and continue later. Only to make sure the API can gracefully evolve later when adding nested objects.